src/caffe/layers/pooling_layer.cpp

```diff
@@ -203,6 +204,9 @@ void PoolingLayer<Dtype>::Forward_cpu(const vector<Blob<Dtype>*>& bottom,
             wstart = max(wstart, 0);
             hend = min(hend, height_);
             wend = min(wend, width_);
+            if (pool_param.dynamic_ave_pool_size() == true) {
```

Comment:
`if (pool_param.dynamic_ave_pool_size())` will suffice. And if you decide to follow my advice from the comment above, it will get even simpler (`if (member_var_)`).